lib/rsolr/client.rb

```diff
@@ -122,6 +122,14 @@ def commit opts = {}
     update opts.merge(:data => builder.commit( commit_attrs ))
   end
 
+  # soft commit
+  #
+  # https://lucene.apache.org/solr/guide/updatehandlers-in-solrconfig.html#commit-and-softcommit
+  #
+  def soft_commit opts = {}
+    commit(opts.merge params: { softCommit: true })
```

Comment:
`softCommit` doesn't appear to be an option any more for a commit update. I think the way to do a soft commit is via `waitSearcher: true`

The docs linked in the comment are for solr configuration, but the functionality in the code uses a POST update. See
: https://solr.apache.org/guide/uploading-data-with-index-handlers.html#commit-and-optimize-during-updates for options at update time.

Reply:
@hackartisan I think softCommit is still a thing - there's just also hard commits where you don't open a searcher.

Reply:
@tpendragon yes I see now. @giteshnandre sorry for my mis-read of the documentation.

Do either of you know how soft commits interact with searcher initialization? I'm curious if a soft commit update blocks until a new searcher is initialized. Neither of these 2 documentation pages mention both.

Reply:
Soft commit definitely blocks until a new searcher is initialized.